mne/io/nirx/tests/test_nirx.py

```diff
@@ -151,7 +151,7 @@ def test_nirsport_v2():
         np.diff(raw.annotations.onset), [2.3, 3.1], atol=0.1)
 
     mon = raw.get_montage()
-    assert len(mon.dig) == 43
+    assert len(mon.dig) == 27
```

Comment:
can you clarify why this is changing?

Reply:
In short, this PR changes the `get_montage` function to return the location of the optodes/sensors. Whereas it previously returned the channels. In other modalities the sensors are typically one-to-one with the channels. But in fNIRS you can have many more channels than sensors. More details below...

fNIRS has sources and detectors (optodes). There can be multiple channels between a single source and detector, with different frequencies of light Also, there can be multiple channels amongst different sources and detectors. So you can have many more channels than optodes.

Traditionally, users have specified their montages by reporting the locations of the sources and detectors. Rather than each channel. For a simple example of 2 sources and two detectors this typically looks something like...
```text
S1     x    y    x
S2     x    y    x
D1     x    y    x
D2     x    y    x
```

Then, for fNIRS data `set_montage()` calls `_set_montage_fnirs()` (see below) to set the source and detector position (and calculate the midpoint as the "channel") for each source-detector pairing based on the above information.

https://github.com/mne-tools/mne-python/blob/9f833658527a2c06ba833bffee33c9ba3a4a528a/mne/channels/montage.py#L859-L867

However, before this PR, `get_montage()` would return the location of each source-detector pair midpoint, so it was not compatible with `set_montage()`. This PR changes the returned values to follow the format above (more precisely, it returns a dig object). As there are typically more channels than optodes, that is why the number above is reduced.

So the question is... Do we want to return data in this reduced form (i.e. one value per optode) as implemented in the PR? Or do we want to keep the previous way of returning all channels, and rewrite the `set_montage_fnirs()` code? 

I favour this approach, because users typically digitise the location of the sources and detectors (I don't think its even possible to digitise the channel positions, there can be 1000s of them). So it makes sense to me that our data structure mirrors what would be digitised. But this deviates from other data types like EEG. So I welcome further questions or feedback here as we decide the best approach.

Reply:
> So the question is... Do we want to return data in this reduced form (i.e. one value per optode) as implemented in the PR? Or do we want to keep the previous way of returning all channels, and rewrite the set_montage_fnirs() code?

Agreed we should return `S1` and `D1` rather than things like `S1_D1 760` or whatever